File: src/main/java/common/SystemPropertyReader.java
package common;

import common.exceptions.MissingSystemPropertyException;

public class SystemPropertyReader {
    public static String readSystemProperty(String propertyName) throws MissingSystemPropertyException {
        String propertyValue = System.getProperty(propertyName);

        if (propertyValue == null) {
            throw new MissingSystemPropertyException("System property '" + propertyName + "' is not defined.");
        }

        return propertyValue;
    }
}

File: src/main/kotlin/common/SystemPropertyReader.kt
package common

import common.exceptions.MissingSystemPropertyException

object SystemPropertyReader {
    @Throws(MissingSystemPropertyException::class)
    fun readSystemProperty(propertyName: String): String {
        val propertyValue = System.getProperty(propertyName)
                ?: throw MissingSystemPropertyException("System property '$propertyName' is not defined.")

        return propertyValue
    }
}

